sui/ibc/sources/ibc.move

```diff
@@ -1688,585 +1688,1329 @@ module ibc::ibc {
-    // //     channel::set_state(&mut channel, CHAN_STATE_OPEN);
-    // //     channel::set_connection_id(&mut channel, connection_id);
-    // //     channel::set_counterparty_channel_id(&mut channel, 1); // Set counterparty channel ID
-    // //     ibc_store.channels.add(channel_id, channel);
-
-    // //     // Prepare packet data
-    // //     let sequence = 1;
-    // //     let packet = packet::new(
-    // //         sequence,
-    // //         channel_id,
-    // //         b"Test data",
-    // //         100, // Timeout height
-    // //         1_000_000_000 // Timeout timestamp
-    // //     );
-    // //     let packets = vector::singleton(packet);
-
-    // //     // Call process_receive
-    // //     let proof = b"valid_proof"; // Mock proof
-    // //     let proof_height = 100;
-    // //     let acknowledgement = b"";
-
-    // //     let commitment_key =
-    // //             commitment::batch_receipts_commitment_key(
-    // //                 packet::destination_channel_id(&packet),
-    // //                 commitment::commit_packet(&packet)
-    // //             );
-                
-    // //     let mut clock = clock::create_for_testing(&mut ctx);
-    // //     clock.set_for_testing(99);
-
-    // //     process_receive(
-    // //         &mut ibc_store,
-    // //         &clock,
-    // //         packets,
-    // //         proof_height,
-    // //         proof,
-    // //         false, // No intent
-    // //         acknowledgement
-    // //     );
-
-    // //     assert!(ibc_store.commitments.contains(commitment_key), E_PACKET_NOT_RECEIVED);
-    // //     let commitment = ibc_store.commitments.borrow(commitment_key);
-    // //     assert!(commitment == COMMITMENT_MAGIC, 0);
-
-
-    // //     clock::destroy_for_testing(clock);
-    // //     test_scenario::return_shared(ibc_store);
-    // //     test_case.end();
-    // // }
+        test_scenario::return_shared(clk2);
+        assert!(ibc.commitments.contains(commitment::client_state_commitment_key(1)), 1);
+
+        t.next_tx(@0x0);
+        // Freeze the client
+        ibc.misbehaviour(1, b"evil", @0xAAA);
+
+        t.next_tx(@0x0);
+        // Any op that checks status==0 should now fail
+        // e.g. connection_open_init(1, ..)
+        ibc.connection_open_init(1, 2);
+
+        test_scenario::return_shared(ibc);
+        t.end();
+    }
+
+    #[test]
+    #[expected_failure(abort_code = E_PACKET_HAVE_NOT_TIMED_OUT)]
+    fun test_commit_packet_timeout_early_fail() {
+        let mut t = test_scenario::begin(@0x0);
+        init_for_tests(t.ctx());
+
+        t.next_tx(@0x0);
+        let mut clk0 = clock::create_for_testing(t.ctx());
+        clock::increment_for_testing(&mut clk0, /*millis=*/ 1_000);
+        clock::share_for_testing(clk0);
+
+        t.next_tx(@0x0);
+        let mut ibc = t.take_shared<IBCStore>();
+        ibc.create_client(string::utf8(b"cometbls"), b"cs1", b"cons1", t.ctx()); 
```

Comment:
im seeing this channel opening mechanism is copied everywhere, let's write a helper function here, which will take `ibc` and `test_context` and create a channel. And reuse it when needed.